synapse/config/ratelimiting.py

```diff
@@ -12,28 +12,42 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import attr
+
 from ._base import Config
 
 
+class ratelimiter(object):
+    def __init__(self, config):
+        self.per_second = config.get("per_second", 0.17)
+        self.burst_count = config.get("burst_count", 3.0)
+
+
+@attr.s
+class rclogin(object):
+    _address = attr.ib()
+    _account = attr.ib()
+
+    def __attrs_post_init__(self):
+        self.address = ratelimiter(self._address)
+        self.account = ratelimiter(self._account)
+
+
 class RatelimitConfig(Config):
 
     def read_config(self, config):
         self.rc_messages_per_second = config["rc_messages_per_second"]
         self.rc_message_burst_count = config["rc_message_burst_count"]
 
+        self.rc_registration = ratelimiter(config.get("rc_registration", {}))
+        self.rc_login = rclogin(**config.get("rc_login", {}))
```

Comment:
I don't understand how this works, and generally I think `rclogin` is confusing; I'd be inclined just to have `rc_login_address` and `rc_login_account` properties in the `RateLimitConfig` object.

```
rc_login_config = config.get("rc_login", {})
self.rc_login_address = ratelimiter(rc_login_config.get("address", {}))
```